micro-rdk/src/common/entry.rs

```diff
@@ -205,20 +204,36 @@ pub async fn serve_inner<S>(
         restart,
     )));
 
-    #[cfg(feature = "native")]
     let server_builder = {
+        let http2_port = 12346;
         server_builder.with_http2(
             {
                 let certs = app_client.get_certificates().await.unwrap();
-                let tls_server_config = NativeTlsServerConfig::new(
-                    certs.tls_certificate.as_bytes().to_vec(),
-                    certs.tls_private_key.as_bytes().to_vec(),
-                );
-                let address: SocketAddr = "0.0.0.0:12346".parse().unwrap();
-                let tls = Box::new(NativeTls::new_server(tls_server_config));
-                NativeListener::new(address.into(), Some(tls)).unwrap()
+                let address = SocketAddr::new("0.0.0.0".parse().unwrap(), http2_port);
+                #[cfg(feature = "native")]
+                {
+                    let tls_server_config = NativeTlsServerConfig::new(
+                        certs.tls_certificate.as_bytes().to_vec(),
+                        certs.tls_private_key.as_bytes().to_vec(),
+                    );
+                    let tls = Box::new(NativeTls::new_server(tls_server_config));
+                    NativeListener::new(address.into(), Some(tls)).unwrap()
+                }
+                #[cfg(feature = "esp32")]
+                {
+                    let tls_server_config = Esp32TLSServerConfig::new(
+                        CString::new(certs.tls_certificate)
+                            .unwrap()
+                            .into_bytes_with_nul(),
+                        CString::new(certs.tls_private_key)
+                            .unwrap()
+                            .into_bytes_with_nul(),
```

Comment:
There's probably a more graceful way to say this, but see below because the whole thing may need to be rethought: converting to null-terminated CString here may be a waste of time. It might be nice because then we could have the same form of args as for `NativeTlsServer` and simplify the blocks a bit.